src/binary/ibc-relayer/setup-prometheus.ts

```diff
@@ -0,0 +1,65 @@
+import http from 'http';
+
+import client from 'prom-client';
+
+import { Logger } from '../create-logger';
+
+let initialized = false;
+
+function getMetrics() {
+  return {
+    pollCounter: new client.Counter({
+      name: 'poll_counter',
+      help: 'Poll counter',
+    }),
+  };
+}
+
+type StubbedMetrics = Record<keyof ReturnType<typeof getMetrics>, null>;
+function getStubbedMetrics(): StubbedMetrics {
+  return {
+    pollCounter: null,
+  };
+}
+
+export type Metrics = ReturnType<typeof getMetrics> | StubbedMetrics;
+export function setupPrometheus({
+  enabled,
+  port,
+  logger,
+}: {
+  enabled: boolean;
+  port: number;
+  logger: Logger;
+}) {
+  if (initialized) {
+    throw new Error(
+      `"setupPrometheus" func shouldn't be initialized more than once.`
+    );
+  }
+  initialized = true;
+
+  if (!enabled) {
+    return getStubbedMetrics();
+  }
+
+  client.collectDefaultMetrics({ prefix: 'confio_relayer_' });
```

Comment:
Not sure if we want to collect these? Reference:
- https://github.com/siimon/prom-client#default-metrics
- https://prometheus.io/docs/instrumenting/writing_clientlibs/#process-metrics